src/IssueViz/AnalysisIssueSelectionEvents.cs

```diff
@@ -0,0 +1,76 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.IssueVisualization
+{
+    [Export(typeof(IAnalysisIssueSelectionEvents))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class AnalysisIssueSelectionEvents : IAnalysisIssueSelectionEvents
+    {
+        private IAnalysisIssue selectedIssue;
+        private IAnalysisIssueFlow selectedFlow;
+        private IAnalysisIssueLocation selectedLocation;
+
+        public event EventHandler<IssueChangedEventArgs> SelectedIssueChanged;
+        public event EventHandler<FlowChangedEventArgs> SelectedFlowChanged;
+        public event EventHandler<LocationChangedEventArgs> SelectedLocationChanged;
+
+        public IAnalysisIssue SelectedIssue
```

Comment:
I'm debating if this service should also be responsible for fixing the relationship between the selected items, e.g.:
1. When SelectedIssue changes, then the SelectedFlow should change (either to null or to first flow of the issue)
2. When SelectedFlow changes, then the SelectedLocation should change (either to null or to the first location of the flow). 

WDYT?